src/client.rs

```diff
@@ -295,32 +311,57 @@ where
             Err(_) => return Err(Error::SocketError),
         };
 
-        // Compare server and client hashes.
-        let password_hash = md5_hash_password(&config.user.name, &config.user.password, &salt);
 
-        if password_hash != password_response {
-            debug!("Password authentication failed");
-            wrong_password(&mut write, &config.user.name).await?;
-            return Err(Error::ClientError);
+
+        let mut target_pool: ConnectionPool = ConnectionPool::default();
+        let mut transaction_mode = false;
+
+        if admin {
+            let correct_user = config.general.admin_username.as_str();
+            let correct_password = config.general.admin_password.as_str();
```

Comment:
This is new. Instead of any authenticated user being allowed to access admin db, a specific username/password is required for that. Honestly, I went with that for simplicity